src/Compilers/CSharp/Test/Syntax/Parsing/StatementAttributeParsingTests.cs

```diff
@@ -5512,14 +5512,18 @@ void Goo(int i)
         [Fact]
         public void AttributeOnExpressionStatement_PrefixUnary()
         {
-            var test = UsingTree(@"
-class C
-{
-    void Goo(int i)
-    {
-        [A]++i;
-    }
-}");
+            var test = UsingTree("""
+                class C
+                {
+                    void Goo(int i)
+                    {
+                        [A]++i;
+                    }
+                }
+                """,
+                // (5,14): error CS1002: ; expected
+                //         [A]++i;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "i").WithLocation(5, 14));
```

Comment:
this was an existing error recovery case that i added a few years back (recovering when attributes were put in incorrect places).  It's still an error after this pr, just one that views the above as an invalid operation on a collection-expr, not an invalid attribute on an expression statement.  Trying to detect this case and keep the old parsing isn't worth it IMO.